website/src/models/group.js

```diff
@@ -89,22 +89,15 @@ GroupSchema.pre('remove', function(next) {
   var group = this;
   async.waterfall([
     function(cb) {
-      var invitationQuery = {};
-      var groupType = group.type;
-      //Add an 's' to group type guild because the model has the plural version
-      if (group.type == "guild") groupType += "s";
-      invitationQuery['invitations.' + groupType + '.id'] = group._id;
```

Comment:
So, do you think that the `invitations.party.id` query was causing the problem, or something else?